src/psyclone/f2pygen.py

```diff
@@ -559,11 +559,15 @@ def __init__(self, parent, content):
         fortran_writer = FortranWriter()
         reader = FortranStringReader(fortran_writer(content),
                                      ignore_comments=False)
+        # Set reader as free form, strict
+        reader.set_format(FortranFormat(True, True))
         fparser1_parser = FortranParser(reader, ignore_comments=False)
         fparser1_parser.parse()
 
         # Update the f2pygen AST with the newly parsed fparser1 AST content
-        BaseGen.__init__(self, parent, fparser1_parser.block.content[0])
+        fparser_node = fparser1_parser.block.content[0]
```

Comment:
I think this should just be `fparser_node.parent = parent.root` along with a comment along the lines of "Graft the newly-created parse tree into that used by the rest of the f2pygen AST." We can then remove line 571.